traits/observers/_i_observer.py

```diff
@@ -29,3 +43,142 @@ def __hash__(self):
     def __eq__(self, other):
         """ Return true if this observer is equal to the given one."""
         raise NotImplementedError("__eq__ must be implemented.")
+
+    @property
+    def notify(self):
+        """ A boolean for whether this observer will notify for changes.
+        """
+        raise NotImplementedError("notify property must be implemented.")
+
+    def iter_observables(self, object):
+        """ Yield observables for adding or removing notifiers.
+
+        The notifiers are contributed by this observer's ``get_notifier``
+        and ``get_maintainer`` methods.
+
+        An observer may observe many items, e.g. multiple traits for a given
+        instance of HasTraits, then the observer will yield many items here.
+
+        An observer may also observe just one thing, e.g. a single trait, or an
+        instance of TraitListObject, then the observer may yield just that.
+
+        An observer may also find nothing to be observed and does not need to
+        complain about that, and it will yield nothing here.
+
+        Observers are allowed to raise here if it is appropriate to do so,
+        e.g. to help catch usage errors.
+
+        Parameters
+        ----------
+        object: object
+            Object provided by the ``iter_objects`` methods from another
+            observers or directly by the user.
+            No guarantee can be made about the type of the object.
+            Concrete should do the sanity check where appropriate.
+
+        Yields
+        ------
+        IObservable
+
+        Raises
+        ------
+        ValueError
+            If the given object cannot be handled by this observer.
```

Comment:
This Raises section is not part of the interface. Implementation is allowed to raise whatever type of exceptions they like.  See the `try-except-Exception-re-raise` in `_AddOrRemoveNotifier.__call__`. I will remove this section from the docstring.